src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/Grid.cs

```diff
@@ -297,6 +318,25 @@ public RowDefinitionCollection RowDefinitions
 
                 return (_data.RowDefinitions);
             }
+            set
+            {
+                _data ??= new ExtendedData();
+                _data.RowDefinitions ??= new RowDefinitionCollection(this);
+                _data.RowDefinitions.Clear();
+                if (value == null)
+                {
+                    return;
+                }
+                foreach (RowDefinition rowDef in value)
+                {
+                    if (rowDef.Parent != null)
```

Comment:
this throws when the definition is already the correct owner/parent

e.g. `grid.ColumnDefinitions = grid.ColumnDefinitions` would throw and shouldn't

Reply:
Ah except you clear everything first and then readd it.